components/brave_rewards/resources/shared/components/need_update_browser_alert_view.tsx

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this file,
+* You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+
+import { LocaleContext } from '../lib/locale_context'
+
+import { AlertCircleIcon } from 'brave-ui/components/icons'
+
+interface Props {
+  drawIcon: boolean
+}
+
+function DrawIcon () {
+  return (
+    <div className='need-update-browser-icon'>
+      <AlertCircleIcon />
+    </div>
+  )
+}
+
+export function NeedUpdateBrowserAlertView (props: Props) {
```

Comment:
It seems that both the styling and layout of this component are context-dependent, and that it really just abstracts over the string keys. Also, it looks like there is only one callsite for each "app".

Since that's the case, I would suggest not introducing a shared component here. Instead, I would just inline the layout/style elements and the icon (for the rewards pages) as needed. In that case, we shouldn't need CSS class names; we can use style elements.

Reply:
fixed